cherryontech/unmatched
# #UM-18 Add working navigation links to buttons in nav bar and style to the buttons

## Description
 This PR adds working navigation links to the buttons in the navigation bar. It also styles them to better match the design file. I'm still working on figuring out how to keep the button background filled when on the respective page


## Related Tickets & Documents
Resolves #UM-18 (with one main lingering acceptance criteria)


## Acceptance Criteria

- [x] The navigation bar is consistently present at the top of every page on the website.
- [x] The navigation bar includes clearly labeled links/buttons for primary sections/pages such as "About Us," and "Resources".
- [x] The "Unmatched" logo is prominently displayed in the header along with the navigation bar.
- [x] Clicking on the "Unmatched" logo navigates the user back to the homepage ("About Us").
- [x] Hovering over a navigation link/button provides a visual indication (e.g., underline, color change) signaling it's interactive.
- [x] Clicking on a navigation link/button takes the user to the corresponding section/page.
- [ ] The currently active section/page is visually differentiated in the navigation bar with a filled background.


## What type of PR is this? (check all applicable)

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓    | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
| ✓    | 🎨 Style                   |
|     | 🔥 Performance Improvements|
|     | :scroll: Docs              |


## Mobile & Desktop Screenshots/Recordings
https://github.com/cherryontech/unmatched/assets/115492619/77590f0b-548f-4055-8462-85c8f9b7f31e


## What I learned
DaisyUI definitely has a lot more limited customization than Tailwind. Instead of using the built-in button component I ended up using only Tailwind for the nav bar buttons bc it's tricky to customize the hover color and fill color in DaisyUI - especially when mixing that button with a React Router Link.

## Testing steps
1. Switch to this branch with the following command: `git switch feature/UM-18-navbar`
2. Install dependencies with the following command: `npm i`
3. Run the following scripts: `npm run dev` to start Vite's dev build & `npm run tailwind` to start the Tailwind CSS build
4. View the app in browser (the `npm run dev` script will return the url you can see the app at)
5. Click on the different components of the navigation bar. The logo and the 'about us' button should navigate you to the home page. The 'resources' button should navigate you to the professionals page. Ensure everything is working as expected.
6. Inspect the code to see how the navigation links were set up and how the buttons were styled.
7.  Interact with the code and ask questions/make suggestions - especially with tips to help out with the missing acceptance criteria.



## [optional] What gif best describes this PR or how it makes you feel?
![bunny](https://media.giphy.com/media/l4JyJHAF8blvfplf2/giphy.gif)
